### DISCUSSION TITLE
The way the app currently loops

### DISCUSSION THREAD

**Discussion by tosamyng** [Nain57/Smart-AutoClicker#229] [General Feature Requests]
Is it possible for you to make the app instead of looping the whole event list trying to find a condition that matches. So to only check for a specific event and only when that one is cleared you move on to the next event. So for example I could give it these instruction:

event1
event2 -> event3

So at the start it will just like now try and check to see if event1 or event2 have any matching condition to execute. If it executes event1 and after its done it will go back to keep looking for event1 and event2 condition just like current app. 

But if it finds a condition for event2 and runs it, it will then only look for the condition that fulfil event3 and nothing else.

The idea is that alot of the actions I want the app to take I know for sure is coming in a predictable order so I want to reduce the app searching for stuff that I know wont happen to try and save cpu power and therefore electric bill.

I hope you can implement my idea and if you dont understand something I explain let me know and ill try and explain it in a different way!

**Comment by Nain57** [comment]
This is already possible by using the new "Change event state" action available with version 2.0.0
